### DISCUSSION TITLE
[Question]: Unexpected Token Usage with Anthropic API

### DISCUSSION THREAD

**Discussion by PogChamp-Peter** [danny-avila/LibreChat#4825] [Q&A]
### What is your question?

I'm using Anthropic's API to access Claude Sonnet 20240620 through LibreChat [Running via npm], and I've run into a weird issue with token usage. I made a simple query just to check how many tokens I'd consume:

![Screenshot (26367)](https://github.com/user-attachments/assets/5a7d5e10-753f-49ca-b935-1db03ee0070f)

When I checked the Anthropic dashboard, the input token count was way too high.

![T_0](https://github.com/user-attachments/assets/03d27c66-cbfb-4f94-b95b-496222880226)

This is the first conversation I’ve ever had with the API, and I’m sure the API key isn’t being used anywhere else. The timestamps match exactly when I sent the query to Sonnet3.5, so it’s not a coincidence.

What’s even stranger is that the dashboard shows token usage for Haiku as well, something I never accessed or even attempted to use. And also the weird HTTP and Streaming diff btw Sonnet and Haiku. 

To rule out caching as the culprit, I disabled it and ran another query in Chat 2. Same result, absurdly high input tokens.

![Screenshot (26369)](https://github.com/user-attachments/assets/17938575-437a-4a48-9e21-4aa1c040ce1f)

I’m not using any prompts or custom instructions. 

![Screenshot (26368)](https://github.com/user-attachments/assets/4cba7b42-1661-4315-97cb-6b3d18f586fa)

Everything is running on the default settings in LibreChat. Any idea what could be causing this massive token discrepancy or how to fix it? I'd appreciate any insight because this behavior doesn't make sense based on my usage.

Thanks in advance!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Haiku is used for creating titles at the start of the conversation. There is a bit of prompting involved for the title, you can also disable titling.

**Reply by PogChamp-Peter** [reply]
Oh okay, that makes sense for Haiku being called :) 

But, I still don't get why Sonnet's using way too much input tokens, when my input roughly amounts to 10-20 tokens... 

Any other option(s) that might affect the input tokens usage?

**Reply by PogChamp-Peter** [reply]
I deleted the librechat.yaml config file and tried again, and it's the same issue, except the request got sent twice:

![Screenshot (26371)](https://github.com/user-attachments/assets/01c07248-325f-4859-be5f-f9d71d85f4fa)

It literally wasted 11k tokens in total for no reason...

![Screenshot (26372)](https://github.com/user-attachments/assets/3eeeb56d-bfc6-475d-9a89-f38ba8923f2f)

I don't know who's at fault here. Is it Anthropic or some extra feature of LibreChat? Can somebody please help?

**Reply by fuegovic** [reply]
Look if you have `Artifacts` enabled. When active it will send a big prompt to the LLM with instructions about its usage

![image](https://github.com/user-attachments/assets/7fa26f77-cfe5-4b4f-b17b-68e2f67b07a4)

**Reply by PogChamp-Peter** [reply]
Is there a way to check the number of input tokens that would be sent before actual sending the input? I wanna absolutely make sure to not end up using another 2k, trying to test if it works or not; 

And, on a related side note, not to sound rude, but why is this enabled by default? Atleast, I would have preferred a pop-up telling you what it is, and that it would cost u extra tokens, cause Anthropic tokens are expensive.

**Comment by PogChamp-Peter** [comment]
I tried the same prompt in big-agi.com and it worked just as expected:

![Screenshot (26374)](https://github.com/user-attachments/assets/c220d52e-7a8b-4596-a3d0-70f87184ef58)

The issue definitely seems to be with LibreChat; I'm not sure if it's a bug or some enabled-by-default feature...
